### PR TITLE
chore(UIForms): enhance uischema condition format

### PR DESCRIPTION
**Pull request by rmannibucau**
**What is the problem this PR is trying to solve?**

Allow to have N conditions on a property and evaluate them as an OR and not an AND

**What is the chosen solution to this problem?**

Since condition is a talend component kit object for now we change the format to a more flexible one

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/UIForm/README.md [hunk 1]
```diff
@@ -283,13 +283,17 @@ Those validations change the `errors` object accordingly.
...
 It is possible to render parts of the forms defined in uiSchema, depending on properties values.
-The uiSchema accepts a `conditions` property, which define all conditions to match to be rendered.
+The uiSchema accepts a `condition` property, which defines recursively all conditions to match to be rendered.
 
 | UISchema conditions property | Description |
 |---|---|
-| conditions[] | Each item defines a condition to meet to be rendered. All conditions must be met. |
-| conditions[].path | Define the path of the formData to test. This supports json path, dot notation, of an array of key. |
-| conditions[].values | Defines all the possible values. If the formData to test is equals to one of those, the condition is met. |
+| condition | Each item defines a condition to meet to be rendered. All conditions must be met. |
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/README.md]: It's only 1 condition now

File: packages/forms/src/UIForm/Widget/condition.js [hunk 2]
```diff
@@ -16,12 +16,31 @@ function getValue(value, strategy) {
...
+	if (!condition.children || condition.children.length === 0) {
+		return true;
+	}
+	const evaluator =
+		(condition.childrenOperator || 'AND').toUpperCase() === 'AND'
+			? Array.prototype.every
+			: Array.prototype.some;
+	return evaluator.call(
+		condition.children,
+		cond =>
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/Widget/condition.js]: `cond => shouldRender(properties, cond)`

File: packages/forms/src/UIForm/fields/MultiSelectTag/__snapshots__/MultiSelectTag.component.test.js.snap [hunk 3]
```diff
@@ -40,7 +39,7 @@ exports[`MultiSelectTag field should render MultiSelectTag 1`] = `
       position="left"
       readOnly={false}
       searching={false}
-      searchingText="Searching for matches..."
+      searchingText="Searching for matches…"
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fields/MultiSelectTag/__snapshots__/MultiSelectTag.component.test.js.snap]: The changes in snapshot are weird. Have you compiled all the repo ?

File: packages/forms/src/UIForm/Widget/condition.js (hunk 2 above)
Comment (rmannibucau) [reply packages/forms/src/UIForm/Widget/condition.js]: causes a lint error :(

File: packages/forms/src/UIForm/fields/MultiSelectTag/__snapshots__/MultiSelectTag.component.test.js.snap (hunk 3 above)
Comment (rmannibucau) [reply packages/forms/src/UIForm/fields/MultiSelectTag/__snapshots__/MultiSelectTag.component.test.js.snap]: fixed, component was not up to date

**Comment by rmannibucau** [comment]
Added a breaking change entry and addressed comments, thanks for the review

**Pull request merged**